tests/test_flows.py

```diff
@@ -380,43 +394,69 @@ def test_channel_selection(self):
+
+        ocrm_line = 0
+        result, ocrm_line, mid = self.parse_log(self.gateway, "controller",
+                                                r'.+OPERATING_CHANNEL_REPORT_MESSAGE,\smid=(\d+)',
+                                                ocrm_line)
+        if (result):
+            self.debug("Confirming ACK_MESSAGE from the controller \
+                with same mid as OPERATING_CHANNEL_REPORT_MESSAGE")
+            self.check_log(self.repeater1, "agent_wlan0", "ACK_MESSAGE, mid=%s" % mid)
+            self.check_log(self.repeater1, "agent_wlan2", "ACK_MESSAGE, mid=%s" % mid)
 
-        # Send channel selection request with chnging tx_power_limit 
         # Payload for transmit power (tp)
-        tp20dBm = "0x14"
-        tp21dBm = "0x15"
+        tp20dBm = 0x14
+        tp21dBm = 0x15
 
-        for payload_transmit_power in (tp20dBm,tp21dBm):
-            transmit_power = int(payload_transmit_power,16)
+        for payload_transmit_power in (tp20dBm, tp21dBm):
             self.debug("Send empty channel selection request with changing tx_power_limit")
             self.gateway_ucc.dev_send_1905(
                 self.mac_repeater1,
                 0x8006,
-                tlv(0x8D,0x0007, f'{self.mac_repeater1_wlan0} {payload_transmit_power}'),
-                tlv(0x8D,0x0007, f'{self.mac_repeater1_wlan2} {payload_transmit_power}')
+                tlv(0x8D, 0x0007, '{} 0x{:02x}'.format(self.mac_repeater1_wlan0,
+                    payload_transmit_power)),
+                tlv(0x8D, 0x0007, '{} 0x{:02x}'.format(self.mac_repeater1_wlan2,
+                    payload_transmit_power))
             )
+            time.sleep(1)
 
-            self.check_log(self.repeater1, "agent_wlan0", "tlvTransmitPowerLimit %d" % transmit_power)
-            self.check_log(self.repeater1, "agent_wlan2", "tlvTransmitPowerLimit %d" % transmit_power)
-            self.check_log(self.gateway, "controller", "tx_power=%d" % transmit_power)
-            self.check_log(self.gateway, "controller", "OPERATING_CHANNEL_REPORT_MESSAGE")
+            self.check_log(self.repeater1, "agent_wlan0", "CHANNEL_SELECTION_REQUEST_MESSAGE")
+            self.check_log(self.repeater1, "agent_wlan2", "CHANNEL_SELECTION_REQUEST_MESSAGE")
+
+            self.check_log(self.repeater1, "agent_wlan0", "tlvTransmitPowerLimit %d" %
+                           payload_transmit_power)
+            self.check_log(self.repeater1, "agent_wlan2", "tlvTransmitPowerLimit %d" %
+                           payload_transmit_power)
+
+            self.check_log(self.gateway, "controller", "tx_power=%d" % payload_transmit_power)
+            self.check_log(self.gateway, "controller", "CHANNEL_SELECTION_RESPONSE_MESSAGE")
+
+            result, ocrm_line, mid = self.parse_log(self.gateway, "controller",
+                                                    r'.+OPERATING_CHANNEL_REPORT_MESSAGE,\smid=(\d+)',  # noqa: E501
+                                                    ocrm_line)
```

Comment:
In this case, a better way (IMO) to work around the line length problem is to not use visual indent:
```suggestion
            result, ocrm_line, mid = self.parse_log(self.gateway, "controller",
                r'.+OPERATING_CHANNEL_REPORT_MESSAGE,\smid=(\d+)',
                ocrm_line)
```
I *think* flake8 accepts that (it doesn't complain about visual indent if visual indent would make it exceed the line length limit).

Reply:
Turns out that I was wrong about that. Visual indent is always required, but you can work around it like this:
```suggestion
            result, ocrm_line, mid = self.parse_log(
                self.gateway, "controller",
                r'.+OPERATING_CHANNEL_REPORT_MESSAGE,\smid=(\d+)',
                ocrm_line)
```
However, that does *not* improve readability, so let's stick with the noqa after all.